docs/src/modules/components/AppNavDrawer.js

```diff
@@ -451,6 +451,7 @@ export default function AppNavDrawer(props) {
               versionSelector={renderVersionSelector([
                 // DATA_GRID_VERSION is set from the X repo
                 { text: `v${process.env.DATA_GRID_VERSION}`, current: true },
+                { text: 'v5', href: `https://mui.com${languagePrefix}/components/data-grid/` },
```

Comment:
What is this line for? I assume this would be enough:

```suggestion
```

Reply:
If I remove this line users won't have a link for the v5 version.

> MUI X (v5, v6) is using and should be using the master branch of the mono repo for its code and docs infra, so I fail to understand how it has an impact.

It's related to why we can't remove the line above. When accessing https://next.mui.com/x/react-data-grid/ the menu will show `v6-alpha.x`, `v5` and `v4`, and in https://mui.com/x/react-data-grid/ it will contain `v6`, `v5.17.4` and `v4`. MUI X v5 will use the master branch of the monorepo, while v6 will use the next branch because I don't have the v6 version in the master branch of the MUI X repo to use here. It comes from `package.json`.

Reply:
@m4theushw I think that as much as we can, there should be a **single version** of `@mui/monorepo` (docs infra + code infra). The value is that fixes done to it like https://github.com/mui/material-ui/pull/34619 don't need to be cherry-picked on two different versions (next and master).

Now, if we assume that we have this as a constraint for the solution, then the option that seems to be with the most potential is to continue in the direction that we started to serve the docs infra as a standalone product. This means, if we were using https://docusaurus.io/, we wouldn't go to https://docusaurus.io/ to open a PR to ask to update versions. We would use the API it exposes to do such. I think we should do the same.